### ISSUE TITLE
Compress resource image to paper size to reduce file size

### ISSUE THREAD

**Issue by JackMorganNZ** [uccser/cs-unplugged#16]

**Comment by JackMorganNZ** [comment]
I implemented a basic resize with the Python Pillow library using the following code:

```python
MM_TO_PIXEL_RATIO = 3.78

# Resize image to reduce file size
if paper_size == "a4":
    max_pixel_height = 267 * MM_TO_PIXEL_RATIO
elif paper_size == "letter":
    max_pixel_height = 249 * MM_TO_PIXEL_RATIO
(width, height) = base_image.size
if height > max_pixel_height:
    ratio = height / max_pixel_height
    width *= ratio
    height *= ratio
    base_image = base_image.resize((int(width), int(height)), Image.ANTIALIAS)
```

However it's very computationally intensive. It would slow down resource generation considerably for small copies and the Django server threw the `Killed` error at 50 copies.

**Comment by JackMorganNZ** [comment]
Closed in dfab2a2ffecb0220855f4dd373ec08d3302ac522
